calibrators/intrinsic_camera_calibrator/ceres_intrinsic_camera_calibrator/include/ceres_intrinsic_camera_calibrator/fov_residual.hpp

```diff
@@ -0,0 +1,356 @@
+   * @param[in] x The x coordinate in camera frame
+   * @param[in] y The y coordinate in camera frame
+   * @param[in] cx The camera center x coordinate
+   * @param[in] cy The camera center y coordinate
+   * @param[in] fx The focal length x
+   * @param[in] fy The focal length y
+   * @param[in] k1 The radial distortion coefficient k1
+   * @param[in] k2 The radial distortion coefficient k2
+   * @param[in] k3 The radial distortion coefficient k3
+   * @param[in] p1 The tangential distortion coefficient p1
+   * @param[in] p2 The tangential distortion coefficient p2
+   * @param[in] k4 The rational distortion coefficient k4
+   * @param[in] k5 The rational distortion coefficient k5
+   * @param[in] k6 The rational distortion coefficient k6
+   * @param[in] depth The depth of the point
+   * @returns The pixel x and y coordinates
+   */
+  template <typename T>
+  std::pair<T, T> cameraToImage(
+    const T x, const T y, const T cx, const T cy, const T fx, const T fy, const T k1, const T k2,
+    const T k3, const T p1, const T p2, const T k4, const T k5, const T k6, T depth = T(1.0)) const
+  {
+    const T xp = x / depth;
+    const T yp = y / depth;
+    const T r2 = xp * xp + yp * yp;
+    const T d = getRadialDist(xp, yp, k1, k2, k3, k4, k5, k6);
+    const T xy = xp * yp;
+    const T tdx = 2.0 * p1 * xy + p2 * (r2 + 2.0 * xp * xp);
+    const T tdy = 2.0 * p2 * xy + p1 * (r2 + 2.0 * yp * yp);
+
+    const T u = cx + fx * (xp * d + tdx);
+    const T v = cy + fy * (yp * d + tdy);
+
+    return std::make_pair(u, v);
+  }
+
+  /*!
+   * Converts a pixel coordinate to a point (undistorted) in camera coordinates system
+   * @param[in] u The x coordinate of pixel in image coordinate system
+   * @param[in] v The y coordinate of pixel in image coordinate system
+   * @param[in] cx The camera center x coordinate
+   * @param[in] cy The camera center y coordinate
+   * @param[in] fx The focal length x
+   * @param[in] fy The focal length y
+   * @param[in] k1 The radial distortion coefficient k1
+   * @param[in] k2 The radial distortion coefficient k2
+   * @param[in] k3 The radial distortion coefficient k3
+   * @param[in] p1 The tangential distortion coefficient p1
+   * @param[in] p2 The tangential distortion coefficient p2
+   * @param[in] k4 The rational distortion coefficient k4
+   * @param[in] k5 The rational distortion coefficient k5
+   * @param[in] k6 The rational distortion coefficient k6
+   * @param[in] depth The depth of the point
+   * @returns The real world x and y coordinates in camera frame
+   */
+  template <typename T>
+  std::pair<T, T> imageToCamera(
+    const T u, const T v, const T cx, const T cy, const T fx, const T fy, const T k1, const T k2,
+    const T k3, const T p1, const T p2, const T k4, const T k5, const T k6, T depth = T(1.0),
+    const T tol = T(1e-6)) const
+  {
+    T xp = (u - cx) / fx;
+    T yp = (v - cy) / fy;
+
+    for (int i = 0; i < UNDIST_ITERS; i++) {
```

Comment:
Just a comment for now (will check during the review)
Using a high number of iterations (I know you consider early stopping, but the threshold seems strict) will result in a very long chain rule (potential gradient issues)

Reply:
Discussed in person and already addressed in newer commits. Resolving this